monai/transforms/croppad/array.py

```diff
@@ -1260,14 +1305,29 @@ def __call__(self, img: torch.Tensor, mode: str | None = None, **pad_kwargs) ->
                 note that `np.pad` treats channel dimension as the first dimension.
 
         """
-        orig_size = img.shape[1:]
         ret = self.padder(self.cropper(img), mode=mode, **pad_kwargs)
         # remove the individual info and combine
         if get_track_meta():
             ret_: MetaTensor = ret  # type: ignore
-            pad_info = ret_.applied_operations.pop(-1)
-            crop_info = ret_.applied_operations.pop(-1)
-            self.push_transform(ret_, orig_size=orig_size, extra_info={"pad_info": pad_info, "crop_info": crop_info})
+            if not self.lazy_evaluation:
+                pad_info = ret_.applied_operations.pop()
+                crop_info = ret_.applied_operations.pop()
+                orig_size = crop_info.get(TraceKeys.ORIG_SIZE)
+                self.push_transform(
+                    ret_, orig_size=orig_size, extra_info={"pad_info": pad_info, "crop_info": crop_info}
+                )
+            else:
+                pad_info = ret_.pending_operations.pop()
+                crop_info = ret_.pending_operations.pop()
+                orig_size = crop_info.get(TraceKeys.ORIG_SIZE)
+                self.push_transform(
```

Comment:
Do we need push_transform here for lazy mode? We will pop transforms even without applying them.

Reply:
Yes, I think let's keep it for now, unless you have tested it and it works without these